batch.go

```diff
@@ -52,15 +61,24 @@ func (q *Queue) DoChan(op func() error) <-chan error {
 	go func() {
 		<-done
 		ch <- b.err
+		close(ch)
```

Comment:
This was missing

Reply:
well, technically you don't have to close all channels. Closing a channel is useful when a channel is used to send an unbounded amount of messages. It basically says: "I hereby signal to all consumers that I'm done producing".

Now, in case you think that closing the channel "just in case" is good practice because if you're not meant to read from it "wouldn't it be better if it's formally closed?", I beg to differ:

Reading from a closed channel is legal and returns the zero value of the channel.
Thus, if you close a channel that is not meant to be ever read again, you might hide a bug.

It's easier to notice that something is wrong if you get a runtime error like:

```
all goroutines are asleep - deadlock!
```

rather than getting a `nil` error and thinking that everything is alright.

That's also why reading from a nil channel blocks in Go. It might be a bit confusing but "closing" a channel is just a signalling feature; it's as if channels had a second data type through which you can carry a "EOF" signal through.

Reply:
Understood, thanks for the explanation!
I thought I was preventing a leak.
I'll revert this.